src/pipelines/bounding_box_cropper/utils/bounding_box_cropper_processing.py

```diff
@@ -0,0 +1,137 @@
+import os
+
+import cv2
+import numpy as np
+from picsellia import DatasetVersion, Client
+from picsellia.types.enums import TagTarget, InferenceType
+from picsellia_annotations.coco import Annotation
+
+from src import Pipeline
+from src import step
+from src.models.dataset.dataset_context import DatasetContext
+from src.steps.processing.utils.dataset_version_creation_processing import (
+    DatasetVersionCreationProcessing,
+)
+
+
+class BoundingBoxCropperProcessing(DatasetVersionCreationProcessing):
+    def __init__(
+        self,
+        client: Client,
+        input_dataset_context: DatasetContext,
+        label_name_to_extract: str,
+        output_dataset_version: DatasetVersion,
+        datalake_name: str,
+        destination_path: str,
+    ):
+        super().__init__(
+            client=client,
+            output_dataset_version=output_dataset_version,
+            dataset_type=InferenceType.CLASSIFICATION,
+            dataset_description=f"Dataset extracted from dataset version "
+            f"'{input_dataset_context.dataset_version.version}' "
+            f"(id: {input_dataset_context.dataset_version.id}) "
+            f"in dataset '{input_dataset_context.dataset_version.name}' "
+            f"with label '{label_name_to_extract}'.",
+            datalake_name=datalake_name,
+        )
+        self.dataset_context = input_dataset_context
+        self.processed_dataset_context = DatasetContext(
+            dataset_name="processed_dataset",
+            dataset_version=self.output_dataset_version,
+            destination_path=destination_path,
+            multi_asset=None,
+            labelmap=None,
+        )
+        self.label_name_to_extract = label_name_to_extract
+
+    def _process_images(self) -> None:
+        for image_filename in os.listdir(self.dataset_context.image_dir):
+            self._process_image(image_filename)
+
+    def _process_image(self, image_filename: str) -> None:
+        image_filepath = os.path.join(self.dataset_context.image_dir, image_filename)
+        image = cv2.imread(image_filepath)
```

Comment:
I can use PIL instead of cv2 yes. I don't think we shoud to a try/except here because the data validation step is testing the corruption of images. What do you think ?